src/main/java/org/jabref/preferences/JabRefPreferences.java

```diff
@@ -2701,6 +2711,46 @@ public MergeDialogPreferences getMergeDialogPreferences() {
         return mergeDialogPreferences;
     }
 
+    @Override
+    public AiPreferences getAiPreferences() {
+        if (aiPreferences != null) {
+            return aiPreferences;
+        }
+
+        String token = getOpenAiTokenFromKeyring();
+        aiPreferences = new AiPreferences(getBoolean(USE_AI), token);
+
+        EasyBind.listen(aiPreferences.openAiTokenProperty(), (obs, oldValue, newValue) -> storeOpenAiTokenToKeyring(newValue));
+        EasyBind.listen(aiPreferences.useAiProperty(), (obs, oldValue, newValue) -> putBoolean(USE_AI, newValue));
+
+        return aiPreferences;
+    }
+
+    private String getOpenAiTokenFromKeyring() {
+        try (final Keyring keyring = Keyring.create()) {
+            String rawPassword = keyring.getPassword("org.jabref.customapikeys", "openaitoken");
+            Password password = new Password(rawPassword, getInternalPreferences().getUserAndHost());
+            return password.decrypt();
+        } catch (Exception e) {
+            // What to do in this place?
```

Comment:
What does the other code do if there is an error?

Reply:
I looked at other code that creates a keyring and just did what they did